library/src/main/java/com/datatorrent/lib/io/fs/AbstractSingleFileOutputOperator.java

```diff
@@ -39,10 +41,37 @@
   @NotNull
   protected String outputFileName;
 
+  /**
+   * partitionedFileName string format specifier 
+      e.g. fileName_physicalPartionId -> %s_%d 
+   */
+  private String partitionedFileNameformat = "%s_%d";
+
+  /**
+   * Derived name for file based on physicalPartitionId
+   */
+  private transient String partitionedFileName;
+
+  /**
+   * Physical partition id for the current partition.
+   */
+  private transient int physicalPartitionId;
+
+  /**
+   * Initializing current partition id, partitionedFileName etc. {@inheritDoc}
+   */
+  @Override
+  public void setup(OperatorContext context)
```

Comment:
OR you prefer to have explicit flag to this off? But, this flag will be honored only for single partition.